data/metric/metrics_test.go

```diff
@@ -141,6 +141,26 @@ func TestSet_MarshalJSON(t *testing.T) {
 	)
 }
 
+func TestSet_UnmarshalJSON(t *testing.T) {
+	raw := []byte(`{"bar":0,"baz":1,"event_type":"some-event-type","foo":0,"k":"v","quux":"bar"}`)
+
+	var ms Set
+	err := ms.UnmarshalJSON(raw)
```

Comment:
instead of calling the method, it0d better using `json.Unmarshal` to ensure interface is fulfilled